src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -387,26 +387,32 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
 
         private void AddTupleTypeName(INamedTypeSymbol symbol)
         {
-            var tupleType = (TupleTypeSymbol)symbol;
+            var tupleType = symbol as TupleTypeSymbol;
+            if (tupleType == null)
+            {
+                return;
```

Comment:
If we simply return, nothing will be added. I would expect us to at least fallback to the underlying type. Can we get one from INamedTypeSymbol? If not, consider adding an API next to IsTupleType.